src/_stylesheets/_grid.scss

```diff
@@ -1,32 +1,101 @@
+/// - `<PROPERTY_NAME>-desktop` The custom property to us for setting the value from desktop on
+///
+/// @param $propertyName - The name of the custom property (including the leading `--`)
+///
+/// @example scss - Creating a grid with set numbers of columns at different breakpoints
+///   // A configurable block selector, reading the property
+///   .app-grid {
+///     @include responsive-custom-property(--app-grid-columns)
+///     display: grid;
+///     grid-template-columns: repeat(var(--app-grid-columns))
+///   }
+///
+///   // Modifier classes setting the values at different breakpoints
+///   .app-grid--1-2-3 {
+///     --app-grid-columns-mobile: 1;
+///     --app-grid-columns-tablet: 2;
+///     --app-grid-columns-desktop: 3;
+///   }
+///
+///   .app-grid--1-2 {
+///     --app-grid-columns-mobile: 1;
+///     --app-grid-columns-tablet: 2;
+///     // Desktop will inherit from tablet
+///   }
+@mixin responsive-custom-property($propertyName) {
+  #{$propertyName}: var(#{$propertyName}-mobile);
+
+  @include govuk-media-query($from: tablet) {
+    #{$propertyName}: var(
+      #{$propertyName}-tablet,
+      var(#{$propertyName}-mobile)
+    );
+  }
+
+  @include govuk-media-query($from: desktop) {
+    #{$propertyName}: var(
+      #{$propertyName}-desktop,
+      var(#{$propertyName}-tablet, var(#{$propertyName}-mobile))
+    );
+  }
+}
+
 .app-grid {
+  @include responsive-custom-property(--app-grid-columns);
+
   --app-grid-columns-mobile: 1;
   --app-grid-gap: #{$govuk-gutter};
 
   display: grid;
-  grid-template-columns: repeat(var(--app-grid-columns-mobile), 1fr);
+  grid-template-columns: repeat(var(--app-grid-columns), 1fr);
   gap: var(--app-grid-gap);
+  @include govuk-responsive-margin(5, $direction: 'bottom');
+}
+
+.app-grid--reduced-gap {
+  --app-grid-gap: 10px;
+}
+
+.app-grid__cell {
+  --app-grid-cell-vertical-align:;
+  @include responsive-custom-property(--app-grid-cell-span);
+
+  align-content: var(--app-grid-cell-vertical-align);
+  grid-column: var(--app-grid-cell-span, var(--app-grid-cell-span-mobile));
 
   @include govuk-media-query($from: tablet) {
-    --app-grid-columns-tablet: var(--app-grid-columns-mobile);
-    grid-template-columns: repeat(var(--app-grid-columns-tablet), 1fr);
+    --app-grid-cell-span: var(
+      --app-grid-cell-span-tablet,
+      var(--app-grid-cell-span-mobile)
+    );
   }
 
   @include govuk-media-query($from: desktop) {
-    --app-grid-columns-desktop: var(--app-grid-columns-tablet);
-    grid-template-columns: repeat(var(--app-grid-columns-desktop), 1fr);
+    --app-grid-cell-span: var(
+      --app-grid-cell-span-desktop,
+      var(--app-grid-cell-span-tablet, var(--app-grid-cell-span-mobile))
+    );
+  }
+
+  > :last-child {
+    margin-bottom: 0;
   }
 }
 
 @each $alignment in (start, center, end) {
   .app-grid__cell--vertical-align-#{$alignment} {
-    align-content: $alignment;
+    --app-grid-cell-vertical-align: #{$alignment};
```

Comment:
A minor regression from this is that there are quite a few instances of this modifier used without `app-grid__cell` that I added as part of https://github.com/alphagov/govuk-brand-guidelines/pull/132. Probably on me for not adding it when I made them in anticipation of a change like this. These elements now don't have any alignment rules, just css vars that currently aren't being used.

This should be a simple case of either applying the `gridCell` shortcode to them or adding the `app-grid__cell` class. There might even be a case for removing this modifier set altogether.